wallet/wallet.go

```diff
@@ -2343,29 +2344,28 @@ func (w *Wallet) GetTransactions(ctx context.Context, startBlock, endBlock *Bloc
 				txs = append(txs, makeTxSummary(dbtx, w, &details[i]))
 			}
 
+			var block *Block
 			if details[0].Block.Height != -1 {
 				blockHash := details[0].Block.Hash
-				res.MinedTransactions = append(res.MinedTransactions, Block{
+				block = &Block{
 					Hash:         &blockHash,
 					Height:       details[0].Block.Height,
 					Timestamp:    details[0].Block.Time.Unix(),
 					Transactions: txs,
-				})
+				}
 			} else {
-				res.UnminedTransactions = txs
+				block = &Block{
+					Height:       -1,
+					Transactions: txs,
```

Comment:
I'm using a block height of -1 to signal `f()` that the transactions are unmined. I did it like this to reduce the number of arguments to `f`, but can rework into separate `f(minedBlock, unimedTx)` if needed